src/pages/LogOutPreviousUserPage.tsx

```diff
@@ -25,49 +30,66 @@ type LogOutPreviousUserPageProps = LogOutPreviousUserPageOnyxProps & StackScreen
+            Navigation.isNavigationReady().then(() => {
+                // We must call goBack() to remove the /transition route from history
+                Navigation.goBack();
+                Navigation.navigate(ROUTES.HOME);
+            });
+            return;
+        }
 
-            // We need to signin and fetch a new authToken, if a user was already authenticated in NewDot, and was redirected to OldDot
-            // and their authToken stored in Onyx becomes invalid.
-            // This workflow is triggered while setting up VBBA. User is redirected from NewDot to OldDot to set up 2FA, and then redirected back to NewDot
-            // On Enabling 2FA, authToken stored in Onyx becomes expired and hence we need to fetch new authToken
-            const shouldForceLogin = route.params.shouldForceLogin === 'true';
-            if (shouldForceLogin) {
-                const email = route.params.email ?? '';
-                const shortLivedAuthToken = route.params.shortLivedAuthToken ?? '';
-                SessionActions.signInWithShortLivedAuthToken(email, shortLivedAuthToken);
-            }
-        });
+        // We need to signin and fetch a new authToken, if a user was already authenticated in NewDot, and was redirected to OldDot
+        // and their authToken stored in Onyx becomes invalid.
+        // This workflow is triggered while setting up VBBA. User is redirected from NewDot to OldDot to set up 2FA, and then redirected back to NewDot
+        // On Enabling 2FA, authToken stored in Onyx becomes expired and hence we need to fetch new authToken
+        const shouldForceLogin = route.params.shouldForceLogin === 'true';
+        if (shouldForceLogin) {
+            const email = route.params.email ?? '';
+            const shortLivedAuthToken = route.params.shortLivedAuthToken ?? '';
+            SessionActions.signInWithShortLivedAuthToken(email, shortLivedAuthToken);
+        }
+        const exitTo = route.params.exitTo as Route | null;
+        // We don't want to navigate to the exitTo route when creating a new workspace from a deep link,
+        // because we already handle creating the optimistic policy and navigating to it in App.setUpPoliciesAndNavigate,
+        // which is already called when AuthScreens mounts.
+        if (exitTo && exitTo !== ROUTES.WORKSPACE_NEW && !isAccountLoading && !isLoggingInAsNewUser) {
+            Navigation.isNavigationReady().then(() => {
+                // remove this screen and navigate to exit route
+                const exitUrl = NativeModules.HybridAppModule ? Navigation.parseHybridAppUrl(exitTo) : exitTo;
+                Navigation.goBack();
+                Navigation.navigate(exitUrl);
+            });
+        }
 
         // We only want to run this effect once on mount (when the page first loads after transitioning from OldDot)
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []);
+    }, [initialURL, isAccountLoading]);
```

Comment:
This useEffect runs continuously because isAccountLoading is set to true in signInWithShortLivedAuthToken and caused this issue #39831